src/main.rs

```diff
@@ -203,36 +175,41 @@ mod app {
         }
     }
 
-    #[task(priority = 1, local=[dac, pwm], shared=[settings, gpio], capacity=1)]
+    #[task(priority = 1, local=[pwm], shared=[dac, settings, gpio], capacity=1)]
     fn settings_update(mut c: settings_update::Context, settings: Settings) {
         // Verify settings and make them available
-        c.shared.settings.lock(|current| *current = settings);
+        c.shared.settings.lock(|current_settings| {
+            *current_settings = settings;
+            // Limit y_min and y_max values here. Will be incorporated into miniconf response later.
+            current_settings.output_channel.iter_mut().for_each(|ch| {
+                ch.iir.y_max = ch.iir.y_max.clamp(0.0, 2.9999);
```

Comment:
`2.9999` is fragile. Better do it computationally:
`impl From<DacCode> for f32` and use `DacCode(u32::MAX).into()`.